### PR TITLE
Sample for user auth with IAP

### PR DESCRIPTION
**Pull request by engelke**

### CODE REVIEW

File: authenticating-users/app.js [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+'use strict';
+
+const express = require('express');
+const got = require('got')
+const jwt = require('jsonwebtoken');
+
+const app = express();
+
+// Cache externally fetched information for future invocations
+let CERTS;
```
Comment (fhinkel) [inline authenticating-users/app.js]: Only `const` variables have all caps.

File: authenticating-users/app.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+let AUDIENCE;
+
+async function certs() {
+  if (!CERTS) {
+    let response = await got('https://www.gstatic.com/iap/verify/public_key');
+    CERTS = JSON.parse(response.body)
```
Comment (fhinkel) [inline authenticating-users/app.js]: I need to update the linter config to run on this. There's a few `;` missing.

File: authenticating-users/app.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  return {
+    'email': payload.email,
+    'sub': payload.sub,
+  };
+
+}
+
+app.get('/', (req, res) => {
+  const assertion = req.header('X-Goog-IAP-JWT-Assertion');
+  validate_assertion(assertion).then((info) => {
```
Comment (fhinkel) [inline authenticating-users/app.js]: We can use `async/await` here:

```
app.get('/', async (req, res) => {
  ...
  try {
    const info = await validate_assertion();
    res...end();
  } catch (error) {
    res...end();
  }
}

File: authenticating-users/app.js (hunk 2 above)
Comment (engelke) [reply authenticating-users/app.js]: That's what I get for switching languages often. I think I caught and fixed them all.

File: authenticating-users/app.js (hunk 1 above)
Comment (engelke) [reply authenticating-users/app.js]: Fixed.

File: authenticating-users/app.js (hunk 3 above)
Comment (engelke) [reply authenticating-users/app.js]: Fixed.

File: authenticating-users/app.js (hunk 2 above)
Comment (engelke) [reply authenticating-users/app.js]: I think all are fixed now.

File: authenticating-users/app.js (hunk 3 above)
Comment (engelke) [reply authenticating-users/app.js]: Reverted. Can't use await here - not an async function.

**Comment by engelke** [comment]
> We only use caps for `const` variables.

Fixed.

> I'm thinking though that we should make them `const` and initialize them before the first request, i.e., before `app.get()`. Would you like to make that change? Otherwise I'm happy to.

I'll take a stab at it. For some reason, I'd rather have the global variables mediated through a function call, but I can't really justify that.

**Comment by engelke** [comment]
Actually, since initializing them involves asynchronous calls, it seems possible that the response to app.get would be created before those variables were initialized. So I'd prefer to leave the code as is.

**Pull request merged**